primitives/tracing/src/types.rs

```diff
@@ -657,6 +562,49 @@ mod std_features {
 		f(metadata, &values)
 	}
 
+	fn make_hex(inp: &[u8]) -> String {
+		inp.iter().map(|c| format!("{:02x}", c)).collect::<String>()
+	}
+
+	impl WasmValue {
+
+		fn as_value<'a>(&'a mut self) -> Option<&'a dyn Value> {
```

Comment:
I get confused by the naming `as_value` which in turn takes a `mut ref` and "mutates" but hey it might just be me...

Reply:
Yeah, not a fan of this API either, but we can't return `dyn Value` for the internal values without changing them :( ... I was also thinking of "convert_to_value" or something ...